modules/navier_stokes/src/fvbcs/PINSFVStrongBC.C

```diff
@@ -0,0 +1,90 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "PINSFVStrongBC.h"
+#include "NS.h"
+#include "SinglePhaseFluidProperties.h"
+#include "Function.h"
+#include "MfrPostprocessor.h"
+
+registerMooseObject("NavierStokesApp", PINSFVStrongBC);
+
+InputParameters
+PINSFVStrongBC::validParams()
+{
+  InputParameters params = FVFluxBC::validParams();
+  params += INSFVFlowBC::validParams();
+  params.addClassDescription("Computes the residual of advective term using finite volume method.");
+  params.addParam<MaterialPropertyName>(NS::density, NS::density, "The density material property");
+  MooseEnum eqn("mass momentum energy");
```

Comment:
```suggestion
  MooseEnum eqn("mass momentum");
```
I m all for energy being in the enum but you need to add the case handling below if so